pinot-core/src/main/java/org/apache/pinot/core/operator/dociditerators/ScanBasedDocIdIterator.java

```diff
@@ -44,4 +44,13 @@ public interface ScanBasedDocIdIterator extends BlockDocIdIterator {
    * iteration. This method should be called after the iteration is done.
    */
   long getNumEntriesScanned();
+
+  /**
+   * Returns the effective cardinality of the underlying data source
+   */
+
+  default float getEffectiveCardinality(boolean isAndDocIdSet) {
```

Comment:
Why are we passing `isAndDocIdSet`?
Suggest changing it to `double getEstimatedSelectivity()`, as it doesn't really return the cardinality.

Reply:
Because for the iterators with N/A matching items (expression, range, regex) we will need to put them in the back of the sorted order so that they evaluate in the end. And due to that AND wants to sort cardinality high->low while OR wants low->high, we need to handle these two cases differently, i.e. return Float.NEGATIVE_INFINITY for AND, and Float.POSITIVE_INFINITY for OR